### PR TITLE
make TraceNode thread safe

### PR DESCRIPTION
**Pull request by arnej27959**
@bjorncs please review

### CODE REVIEW

File: vespajlib/src/main/java/com/yahoo/concurrent/ThreadRobustList.java [hunk 1]
```diff
@@ -16,7 +16,9 @@
  *
  * @author Steinar Knutsen
  * @author bratseth
+ * @deprecated Not used, will be removed in next major release
  */
+@Deprecated // TODO: Remove on Vespa 9
```
Comment (bjorncs) [inline vespajlib/src/main/java/com/yahoo/concurrent/ThreadRobustList.java:21]:
```suggestion
@Deprecated(forRemoval = true) // TODO: Remove on Vespa 9
```

File: vespajlib/src/main/java/com/yahoo/yolean/concurrent/ThreadRobustList.java [hunk 2]
```diff
@@ -20,8 +20,9 @@
  *
  * @author Steinar Knutsen
  * @author bratseth
- * @since 5.1.15
+ * @deprecated Not used, will be removed in next major release
  */
+@Deprecated // TODO: Remove on Vespa 9
```
Comment (bjorncs) [inline vespajlib/src/main/java/com/yahoo/yolean/concurrent/ThreadRobustList.java:25]:
```suggestion
@Deprecated(forRemoval = true) // TODO: Remove on Vespa 9
```

File: vespajlib/src/test/java/com/yahoo/concurrent/ThreadRobustListTestCase.java [hunk 3]
```diff
@@ -12,8 +12,9 @@
 /**
  * Check we keep the consistent view when reading and writing in parallell.
  *
- * @author <a href="mailto:steinar@yahoo-inc.com">Steinar Knutsen</a>
+ * @author Steinar Knutsen
  */
+@SuppressWarnings("deprecation")
```
Comment (bjorncs) [inline vespajlib/src/test/java/com/yahoo/concurrent/ThreadRobustListTestCase.java:17]:
```suggestion
@SuppressWarnings("removal")
```

File: vespajlib/src/test/java/com/yahoo/yolean/concurrent/ThreadRobustListTestCase.java [hunk 4]
```diff
@@ -21,8 +21,9 @@
 import static org.junit.Assert.fail;
 
 /**
- * @author <a href="mailto:steinar@yahoo-inc.com">Steinar Knutsen</a>
+ * @author Steinar Knutsen
  */
+@SuppressWarnings("deprecation")
```
Comment (bjorncs) [inline vespajlib/src/test/java/com/yahoo/yolean/concurrent/ThreadRobustListTestCase.java:26]:
```suggestion
@SuppressWarnings("removal")
```

**Review by bjorncs — DISMISSED** [review]

**Comment by arnej27959** [comment]
@bjorncs please take another look

**Review by bjorncs — APPROVED** [review]

**Pull request merged**